File: seminar-06/breadth_first_search/java/Bfs.java
import java.util.*;

class Bfs {
    public static boolean[] bfs(int start, List<List<Integer>> graph) {
        int n = graph.size();
        boolean[] visited = new boolean[n];

        Queue<Integer> q = new ArrayDeque<>();
        q.add(start);
        visited[start] = true;

        while (!q.isEmpty()) {
            int v = q.poll();

            for (int to : graph.get(v)) {
                if (!visited[to]) {
                    visited[to] = true;
                    q.add(to);
                }
            }
        }

        return visited;
    }
}

File: seminar-06/breadth_first_search/cpp/bfs.cpp
#include <vector>
#include <queue>
using namespace std;

vector<bool> bfs(int start, const vector<vector<int>>& graph) {
    int n = graph.size();
    vector<bool> visited(n, false);

    queue<int> q;
    q.push(start);
    visited[start] = true;

    while (!q.empty()) {
        int v = q.front();
        q.pop();

        for (int to : graph[v]) {
            if (!visited[to]) {
                visited[to] = true;
                q.push(to);
            }
        }
    }

    return visited;
}